api/api.go

```diff
@@ -18,34 +18,40 @@ import (
 
 type API struct {
 	baseUrl          string
+	defaultPodman    bool
+	hostUser         string
 	httpClient       *http.Client
 	httpStreamClient *http.Client
 	logger           hclog.Logger
 }
 
 type ClientConfig struct {
-	SocketPath  string
-	HttpTimeout time.Duration
+	SocketPath    string
+	HttpTimeout   time.Duration
+	HostUser      string
+	DefaultPodman bool
 }
 
 func DefaultClientConfig() ClientConfig {
 	cfg := ClientConfig{
 		HttpTimeout: 60 * time.Second,
 	}
 	uid := os.Getuid()
-	// are we root?
 	if uid == 0 {
 		cfg.SocketPath = "unix:///run/podman/podman.sock"
 	} else {
-		// not? then let's try the default per-user socket location
 		cfg.SocketPath = fmt.Sprintf("unix:///run/user/%d/podman/podman.sock", uid)
 	}
+	cfg.HostUser = "root"
```

Comment:
I removed the HostUser field from the config block, since it was not used anywhere in the driver. There were also places in the `driver.go` code where I couldn't do better than just setting an empty string. Removing the host user, removes the issue raised :)